src/sage/sat/boolean_polynomials.py

```diff
@@ -340,11 +346,11 @@ def learn(F, converter=None, solver=None, max_learnt_length=3, interreduction=Fa
 
     We construct a simple system and solve it::
 
-       sage: set_random_seed(2300)                      # optional - pycryptosat
-       sage: sr = mq.SR(1,2,2,4,gf2=True,polybori=True) # optional - pycryptosat
-       sage: F,s = sr.polynomial_system()               # optional - pycryptosat
-       sage: H = learn_sat(F)                           # optional - pycryptosat
-       sage: H[-1]                                      # optional - pycryptosat
+       sage: set_random_seed(2300)
+       sage: sr = mq.SR(1, 2, 2, 4, gf2=True, polybori=True)    # optional - pycryptosat, needs sage.modules sage.rings.finite_rings
+       sage: F,s = sr.polynomial_system()       # optional - pycryptosat                # needs sage.modules sage.rings.finite_rings
```

Comment:
could be better to use block scope in view of the number of tags

Reply:
Done in d1c66423cf6, actually using file-level